userenvs/fedora31.json

```diff
@@ -0,0 +1,41 @@
+{
+    "workshop": {
+	"schema": {
+	    "version": "2020.03.02"
+	}
+    },
+    "userenv": {
+	"name": "fedora31",
+	"label": "Fedora 31",
+	"origin": {
+	    "image": "docker.io/library/fedora",
+	    "tag": "31"
+	},
+	"properties": {
+	    "packages": {
+		"type": "rpm",
+		"manager": "dnf"
+	    }
+	}
+    },
+    "requirements": [
+	{
+	    "name": "python36",
+	    "type": "distro",
+	    "distro_info": {
+		"packages": [
+		    "python36"
+		]
+	    }
+	},
+	{
+	    "name": "development-tools",
+	    "type": "distro",
+	    "distro_info": {
+		"groups": [
```

Comment:
I wonder how we indicate to users that some distros don't have a group-install option.  I suppose under properties->packages we could have a field name for install types that lists the types like package, group.

Reply:
I've been thinking about that and the obvious one that comes to mind is Debian based distros.  I think in that situation the workshop.pl code would just treat groups and packages as the same since Debian has the concept of meta packages that just resolve internally to many packages.

Reply:
I also think that there needs to be some assumed intelligence on the user's behalf.  If they write a JSON that tells workshop to install a group called 'X' from a distro that doesn't support groups then where the heck did they get 'X' from?  If the distro doesn't support groups it may just be that the workshop.pl code errors out.  One thing to remember about the schema as it currently stands is that only the userenv JSON files know anything about the distro and it's behavior.  The requirements JSON files are very generic and the requirement sections that are mapped to a userenv are assumed to be compliant.  The JSON/schema validator is not capable (as is) of resolving/checking that.  Problems like that will be handled in workshop.pl.